### ISSUE TITLE
Default Parameter Values being NULL breaks API calls

### ISSUE THREAD

**Issue by samterfa** [MarkEdmondson1234/autoGoogleAPI#4]
First, thank you for these packages!

The default values for each function generated is NULL which generally throws an error "JSON fetch error: Invalid string value: 'NULL'". I can get around this by setting each unneeded parameter to "" instead but some functions have lots of parameters which gets cumbersome. Could they be set by default to ""?

**Comment by MarkEdmondson1234** [comment]
Yes this is a bug I haven't documented yet. 

I fix this by using a helper function that removes all `NULL` from a list, and pass that instead.

Here is my fix I used in a project recently.

I have two helper functions to help with NULLs:

```r
#' A helper function that tests whether an object is either NULL _or_
#' a list of NULLs
#'
#' @keywords internal
is.NullOb <- function(x) is.null(x) | all(sapply(x, is.null))

#' Recursively step down into list, removing all such objects
#'
#' @keywords internal
rmNullObs <- function(x) {
  x <- Filter(Negate(is.NullOb), x)
  lapply(x, function(x) if (is.list(x)) rmNullObs(x) else x)
}


```

Then I use that like in this function for `googleComputeEngineR`:

```r
  url <- sprintf("https://www.googleapis.com/compute/v1/projects/%s/aggregated/machineTypes", 
                 project)
  
  pars <- list(filter = filter, 
               maxResults = maxResults, 
               pageToken = pageToken)

## remove NULL obs as it breaks the API call
  pars <- rmNullObs(pars)

  # compute.machineTypes.aggregatedList
  f <- gar_api_generator(url, 
                         "GET", 
                         pars_args = pars, 
                         data_parse_function = function(x) x)
  f()
```

I will add this fix to `googleAuthR` and regenerate these libraries soon.  Also be aware of another bug I found, the order of the parameters passed into the API can be in the wrong order. ( https://github.com/MarkEdmondson1234/autoGoogleAPI/issues/2 )

**Comment by samterfa** [comment]
Thanks for the fixes and heads-up on the other bug.

**Comment by MarkEdmondson1234** [comment]
This is fixed from googleAuthR 0.5
